src/sbvr-api/permissions.ts

```diff
@@ -70,20 +70,20 @@ export interface PermissionReq {
 	user?: User;
 	apiKey?: ApiKey;
 }
-export const root: PermissionReq = {
+export const root: Readonly<PermissionReq> = Object.freeze({
 	user: {
 		id: 0,
 		actor: 0,
 		permissions: ['resource.all'],
 	},
-};
-export const rootRead: PermissionReq = {
+});
+export const rootRead: Readonly<PermissionReq> = Object.freeze({
```

Comment:
You probably want to use the deep-freeze module to make sure all parts are frozen rather than just the top-level